test/unit/spec/common/cellComponents/filePathWidget-Spec.js

```diff
@@ -196,8 +204,64 @@ define([
                 }).then(() => {
                     // the file path list should be empty
                     const postClickNumberOfRows = $node.find('li.kb-file-path__list_item').length;
-                    expect(postClickNumberOfRows).toEqual(0);
+                    expect(postClickNumberOfRows).toEqual(1);
+                });
+            });
+
+            it('should check for duplicate values when an input is changed', async function () {
+                // this might be more of an integration test?
+                // the first couple selectors depend on info from the input widget DOM structure itself,
+                // so, not exactly black box.
+
+                // We're testing duplicate values of output objects. The validator calls out
+                // to the workspace to see if there's already an object of this name - crashes
+                // seem to occur when it can't get there.
+                Mocks.mockJsonRpc1Call({
+                    url: Config.url('workspace'),
+                    body: /get_object_info_new/,
+                    response: {
+                        data: [[null]],
+                    },
                 });
+                const row1 = this.node.querySelector('li.kb-file-path__list_item:first-child');
+                const row2 = this.node.querySelector('li.kb-file-path__list_item:nth-child(2)');
+                const widgetSelector = 'div[data-parameter="actual_output_object"]';
+                const rowCellSelector = `${widgetSelector} .kb-field-cell__rowCell`;
+                const inputSelector = `${widgetSelector} input.form-control`;
+                const dupMsgSelector = `${rowCellSelector} .kb-field-cell__message_panel__duplicate`;
+
+                // wait on the inputs to get rendered
+                const input1 = await TestUtil.waitForElement(row1, inputSelector);
+                const input2 = await TestUtil.waitForElement(row2, inputSelector);
+
+                // wait for the error class to get put on the main widget after a duplicate
+                // value is set
+                await TestUtil.waitForElementState(
+                    this.node,
+                    () => {
+                        return row1
+                            .querySelector(rowCellSelector)
+                            .classList.contains('kb-field-cell__error_message');
+                    },
+                    () => {
+                        input2.setAttribute('value', input1.getAttribute('value'));
+                        input2.dispatchEvent(new Event('change'));
+                    }
+                );
+                expect(row1.querySelector(rowCellSelector).classList).toContain(
```

Comment:
you could refactor this so you're not repeating the same expectations for `row1` and `row2`